src/domain/portfolio/entities/portfolio.entity.ts

```diff
@@ -84,7 +85,7 @@ export const ProtocolPositionType = [
 ] as const;
 
 export const ProtocolPositionsSchema = z.record(
-  z.enum(ProtocolPositionType),
+  z.enum([...ProtocolPositionType, 'UNKNOWN']).catch('UNKNOWN'),
```

Comment:
I think we should add logging here but I'm not sure on the best approach as we'd need to inject the logging service into the entity. To keep the footprint focused, I will address this in a follow up PR.